src/git-repo-research-mcp-server/awslabs/git_repo_research_mcp_server/indexer.py

```diff
@@ -0,0 +1,740 @@
+                rel_path = os.path.relpath(root, repo_path)
+                if rel_path == '.':
+                    rel_path = ''
+
+                # Create the corresponding directory in the target
+                target_dir = os.path.join(repo_files_path, rel_path)
+                os.makedirs(target_dir, exist_ok=True)
+
+                # Copy all files in this directory
+                for file in files:
+                    source_file = os.path.join(root, file)
+                    target_file = os.path.join(target_dir, file)
+                    try:
+                        shutil.copy2(source_file, target_file)
+                        copied_files += 1
+                    except Exception as e:
+                        logger.warning(f'Error copying file {source_file}: {e}')
+
+            logger.info(f'Copied {copied_files} files to {repo_files_path}')
+
+            # Create FAISS index using LangChain
+            logger.info('Creating FAISS index with LangChain')
+            if ctx:
+                ctx.info('Creating FAISS index...')
+                ctx.report_progress(70, 100)  # 70% progress - starting index creation
+
+            embedding_function = self.embedding_generator.bedrock_embeddings
+
+            # Debug: Print document count and first document
+            logger.info(f'Document count: {len(documents)}')
+            if documents:
+                logger.info(f'First document content: {documents[0].page_content[:100]}...')
+                logger.info(f'First document metadata: {documents[0].metadata}')
+
+            # Create the FAISS index
+            logger.info('Creating FAISS vector store from documents')
+            if ctx:
+                ctx.info('Generating embeddings and creating vector store...')
+                ctx.report_progress(75, 100)  # 75% progress - creating vector store
+
+            vector_store = FAISS.from_documents(
+                documents=documents,
+                embedding=embedding_function,
+            )
+
+            # Debug: Print vector store info
+            logger.info(
+                f'Vector store created with {len(vector_store.docstore._dict)} documents'  # pyright: ignore[reportAttributeAccessIssue]
+            )
+
+            # Save the index without pickle
+            logger.info(f'Saving index to {index_path}')
+            if ctx:
+                ctx.info(f'Saving index to {index_path}')
+                ctx.report_progress(85, 100)  # 85% progress - saving index
+
+            save_index_without_pickle(vector_store, index_path)
+
+            # Verify the saved index
+            logger.info('Verifying saved index')
+            try:
+                test_store = load_index_without_pickle(index_path, embedding_function)
+                logger.info(
+                    f'Loaded index contains {len(test_store.docstore._dict)} documents'  # pyright: ignore[reportAttributeAccessIssue]
```

Comment:
Same question as above

Reply:
thank you, added helper extension to check